### PR TITLE
[IMP] default_warehouse_from_sale_team: make abstract model more general T#79454

### PR DESCRIPTION
**Pull request by xmglord**
In v14 when a record is created, the fields with default values were not in the vals passed to the create method but since v15 this behaviour changed so this commit makes the validation of the abstract model more robust considering also the default value.

### CODE REVIEW

**Comment by xmglord** [comment]
@luisg123v please review, this is related to the stock manual transfer MR in [here](https://git.vauxoo.com/vauxoo/typ/-/merge_requests/1139)

File: default_warehouse_from_sale_team/models/default_warehouse_mixin.py [hunk 1]
```diff
@@ -47,7 +47,9 @@ def create(self, vals):
 
     def _get_salesteam_from_vals(self, vals):
         """Determine sales team from creation values"""
-        if "name" in vals:
+        name_field = self._fields["name"]
+        default_name = name_field.default(self) if callable(name_field.default) else name_field.default
+        if vals.get("name", default_name) != default_name:
```
Comment (luisg123v) [inline default_warehouse_from_sale_team/models/default_warehouse_mixin.py:52]: This doesn't seem correct to me. Why is this necessary?

**Review by luisg123v — CHANGES_REQUESTED** [review]
CC @teposteaj

File: default_warehouse_from_sale_team/models/default_warehouse_mixin.py (hunk 1 above)
Comment (xmglord) [reply default_warehouse_from_sale_team/models/default_warehouse_mixin.py:52]: As discussed earlier I'll close this PR since [this new PR](https://github.com/Vauxoo/addons-vauxoo/pull/1664) solve the issue of why this was created.
